hnn_core/cell.py

```diff
@@ -220,6 +218,64 @@ def _connect(self, gid, gid_dict, pos_dict, p, type_src, name_src,
                     self.parconnect_from_src(
                         gid_src, nc_dict, postsyn))
 
+    # one parreceive function to handle all types of external parreceives
+    def parreceive_ext(self, type, gid, gid_dict, pos_dict, p_ext):
+        """Connect cell to external input."""
+        if self.celltype not in p_ext.keys():
+            return
+
+        gid_src = gid + gid_dict[type][0]
+
+        nc_dict = dict()
+        nc_dict['ampa'] = {
+            'pos_src': pos_dict[type][gid],
+            # index 0 for ampa weight
+            'A_weight': p_ext[self.celltype][0],
+            'A_delay': p_ext[self.celltype][2],  # index 2 for delay
+            'lamtha': p_ext['lamtha'],
+            'threshold': p_ext['threshold'],
+            'type_src': type
+        }
+
+        nc_dict['nmda'] = nc_dict['ampa'].copy()
+        nc_dict['nmda']['A_weight'] = p_ext[self.celltype][1]
+
+        if type.startswith(('evprox', 'evdist')) or type == 'extpois':
+            receptors = ['ampa', 'nmda']
+        if type == 'extgauss':
+            receptors = ['ampa']
+
+        # NEW: note that default/original is 0 nmda weight
+        # for the proximal dends
+        for receptor in receptors:
+            if nc_dict[receptor]['A_weight'] > 0.:
+                self._connect_feed_at_loc(
+                    feed_loc=p_ext['loc'], receptor=receptor,
+                    gid_src=gid_src, nc_dict=nc_dict[receptor],
+                    nc_list=self.ncfrom_unique)
+
+    # receive from common inputs
+    # XXX check NetCon connections for proximal inputs with zero weights
+    def parreceive(self, gid, gid_dict, pos_dict, p_ext):
```

Comment:
Should we go ahead and name this `receive_common_feeds` and the other one `receive_unique_feeds`, or shall we save the naming of functions for a later PR since we'll probably keep refactoring a lot of this stuff?

Reply:
Also, I'm not sure I understand the need to check NetCon connections for proximal inputs with zero weights. Could you explain?

Reply:
Yeah, I was wondering about the same. The check was there for the NMDA receptor in extpois. I think it should be the same for all feeds -- whether to keep the check or not. Keeping the check in there felt safer than the other way round. Would you disagree? 

I believe what I did should be correct because this changed the logic for evoked inputs (no check for 0 weights before this PR), yet the output did not change.

Reply:
> Should we go ahead and name this receive_common_feeds and the other one receive_unique_feeds, or shall we save the naming of functions for a later PR since we'll probably keep refactoring a lot of this stuff?

yes, I like the idea for the naming but let's save it for the next PR since I'm trying to refactor all this into a single function in the next PR, so it might become redundant anyway.

Reply:
I just saw #164. Scrap my comment above about renaming `parreceive`. Now is probably not the best time.

Reply:
> Yeah, I was wondering about the same. The check was there for the NMDA receptor in extpois. I think it should be the same for all feeds -- whether to keep the check or not. Keeping the check in there felt safer than the other way round. Would you disagree?

I agree that it should be the same for all feeds, however, we may not want to change the nrn network connectivity based on connection weights being zero. I think we might benefit in the future from instantiating the same nrn network every time and also adding a test to confirm that all connections that are specified in the param file have been translated into nrn correctly. Given your observation below and that a weight of 0 for a given receptor and no instantiated nrn network connection for that receptor are logically equivalent, this allows us to ensure that what we set out to create is created.

> I believe what I did should be correct because this changed the logic for evoked inputs (no check for 0 weights before this PR), yet the output in the example did not change.

Yes. This is obviously because a weight of 0 for a given receptor and no instantiated nrn network connection for that receptor are logically equivalent.

Reply:
In a nutshell, I think the feed objects and nrn network itself should be responsible for the impact of a 0-weighted connection.

Reply:
okay agreed. Let me try to see if it works without the check!

Reply:
okay I pushed commit fbdda25. Seems like nothing changes if I remove the check, so we're good. I also added an entry in the Bug section of `whats_new` so that folks don't try to disable connections by setting the weights to 0. I know this is what people used to the GUI are doing.